humblegen/src/rust/service_server.rs

```diff
@@ -119,7 +121,7 @@ pub fn render_services<'a, I: Iterator<Item = &'a ast::ServiceDef>>(
             /// and `root="/api"` will expose
             /// * handler method `fn bar() -> i32` at `/api/bar` and
             /// * handler method `fn baz() -> String` at `/api/baz`
-            pub fn add(mut self, root: &str, handler: Handler) -> Self {
+            pub fn add<Context: Default + Sized + Send + Sync>(mut self, root: &str, handler: Handler<Context>) -> Self {
```

Comment:
If we are creating a new context for each request, why does it need to be sync?

Reply:
Because it's a requirement of the trait's associated type.
And the associated type requires `sync` because, according to the Rust compile (not to myself), we are moving the context across an await point.